frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-storageclass-selection.spec.ts

```diff
@@ -59,94 +47,103 @@ describe('Add capacity using multiple storage classes', () => {
+      const { name: scName } = testNoProvisionerSC.metadata;
+      allWorkerNodes.forEach((node, id) => {
+        const nodeName = node.metadata.name;
+        cy.exec(`echo '${JSON.stringify(getTestPV(id, nodeName, scName))}' | kubectl apply -f -`);
+      });
     });
     commonFlows.navigateToOCS();
     cy.byLegacyTestID('horizontal-link-Storage Cluster').click();
   });
+
+  beforeEach(() => {
+    fetchStorageClusterJson().then((res) => {
+      const json: K8sResourceKind = JSON.parse(res.stdout);
+      beforeCapacityAddition.deviceSets = json.spec.storageDeviceSets;
+      beforeCapacityAddition.uid = json.metadata.uid;
+    });
+  });
+
   after(() => {
     cy.exec(`echo '${JSON.stringify(testEbsSC)}' | kubectl delete -f -`);
     cy.exec(`echo '${JSON.stringify(testNoProvisionerSC)}' | kubectl delete -f -`);
-    cy.exec(`echo '${JSON.stringify(testPersistentVolume1)}' | kubectl delete -f -`);
-    cy.exec(`echo '${JSON.stringify(testPersistentVolume2)}' | kubectl delete -f -`);
-    cy.exec(`echo '${JSON.stringify(testPersistentVolume3)}' | kubectl delete -f -`);
+    cy.exec('oc get nodes -o json').then((res) => {
+      const nodes = JSON.parse(res.stdout);
+      const allWorkerNodes = nodes.items.filter((node) => getNodeRole(node) === 'worker');
+      const { name: scName } = testNoProvisionerSC.metadata;
+      allWorkerNodes.forEach((node, id) => {
+        const nodeName = node.metadata.name;
+        cy.exec(`echo '${JSON.stringify(getTestPV(id, nodeName, scName))}' | kubectl delete -f -`);
+      });
+    });
     cy.logout();
   });
 
-  describe('Add capacity with a new storage class having EBS as provisioner', () => {
+  it('Add capacity with a new storage class having EBS as provisioner', () => {
     const { name: scName } = testEbsSC.metadata;
     const iAndD: IndexAndDeviceSet = { index: 0, deviceSets: [] };
-    before(() => {
-      fetchStorageClusterJson().then((res) => {
-        const json: K8sResourceKind = JSON.parse(res.stdout);
-        beforeCapacityAddition.deviceSets = json.spec.storageDeviceSets.length;
-        addCapacity(json.metadata.uid, scName);
-        fetchStorageClusterJson().then((newRes) => {
-          withJSONResult(newRes, scName, iAndD);
-        });
-      });
+    addCapacity(beforeCapacityAddition.uid, scName);
+    fetchStorageClusterJson().then((res) => {
+      withJSONResult(res, scName, iAndD);
     });
-    it('New device set is created', () =>
-      expect(iAndD.deviceSets.length).toBe(beforeCapacityAddition.deviceSets + 1));
-    it('Device count is 1 in the new device set', () =>
-      expect(iAndD.deviceSets[iAndD.index].count).toBe(1));
-    it('Osd portability is enabled in the new device set', () =>
-      expect(iAndD.deviceSets[iAndD.index].portable).toBe(true));
+
+    cy.log('New device set is created').then(() =>
+      expect(iAndD.deviceSets.length).toBe(beforeCapacityAddition.deviceSets.length + 1),
+    );
```

Comment:
```suggestion
    cy.log('Test if new device set is added to the cluster');
    expect(iAndD.deviceSets.length).toBe(beforeCapacityAddition.deviceSets.length + 1);
```
I don't see the need to chain expect statements to a log statement. Let's change it for all instances.

Reply:
for some reason the `it` block is failing if we are using `expect` without chaining it with `.then` !!
They are working fine with `.then` chaining... 
all I could find was it was written in the function's description that `The expect function is used every time you want to test a value. You will rarely call expect by itself.`